packages/components/src/Button/components/GenericButton.tsx

```diff
@@ -0,0 +1,366 @@
+import React, {
+  forwardRef,
+  Ref,
+  useImperativeHandle,
+  useRef,
+  ComponentType,
+  FocusEvent,
+  MouseEvent,
+} from "react"
+import classnames from "classnames"
+import { Badge, BadgeAnimated } from "@kaizen/draft-badge"
+import { LoadingSpinner } from "@kaizen/loading-spinner"
+import { ButtonProps } from "../Button"
+import styles from "./GenericButton.module.scss"
+
+export type CustomButtonProps = {
+  id?: string
+  className: string
+  href?: string
+  disabled?: boolean
+  onClick?: (e: MouseEvent<any>) => void
+  onFocus?: (e: FocusEvent<HTMLElement>) => void
+  onBlur?: (e: FocusEvent<HTMLElement>) => void
+  children?: React.ReactNode
+}
+
+export type ButtonFormAttributes = Pick<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  | "form"
+  | "formAction"
+  | "formMethod"
+  | "formEncType"
+  | "formTarget"
+  | "formNoValidate"
+>
+
+export type GenericProps = {
+  id?: string
+  reversed?: boolean
+  onClick?: (e: MouseEvent) => void
+  onMouseDown?: (e: MouseEvent) => void
+  href?: string
+  newTabAndIUnderstandTheAccessibilityImplications?: boolean
+  disableTabFocusAndIUnderstandTheAccessibilityImplications?: boolean
+  onFocus?: (e: FocusEvent<HTMLElement>) => void
+  onBlur?: (e: FocusEvent<HTMLElement>) => void
+  component?: ComponentType<CustomButtonProps>
+  classNameOverride?: string
```

Comment:
Any chance we can throw this on the container element instead of wherever it is now?

Reply:
Hmm I think keep changes here to a minimum, too many snapshots will go wild if this changes.

Reply:
Really? I thought we never tested `classNameOverride` lol

Reply:
Oh I mean in consumer land, if they were using it.

Reply:
But no one is using this component 😂 The `icon` change is more obvious than the `classNameOverride` if anything